--- gcc/config/i386/cygwin.h~	2002-07-15 04:05:45.000000000 +0000
+++ gcc/config/i386/cygwin.h	2002-10-20 07:25:18.000000000 +0000
@@ -110,3 +110,3 @@
 #ifdef GPLUSPLUS_INCLUDE_DIR
-char cygwin_gplusplus_include_dir[sizeof (GPLUSPLUS_INCLUDE_DIR) + 1] = GPLUSPLUS_INCLUDE_DIR;
+char cygwin_gplusplus_include_dir[sizeof (GPLUSPLUS_INCLUDE_DIR) + 10] = GPLUSPLUS_INCLUDE_DIR;
 #undef GPLUSPLUS_INCLUDE_DIR
@@ -119,3 +119,3 @@
 #ifdef GPLUSPLUS_TOOL_INCLUDE_DIR
-char cygwin_gplusplus_tool_include_dir[sizeof (GPLUSPLUS_TOOL_INCLUDE_DIR) + 1] = GPLUSPLUS_TOOL_INCLUDE_DIR;
+char cygwin_gplusplus_tool_include_dir[sizeof (GPLUSPLUS_TOOL_INCLUDE_DIR) + 10] = GPLUSPLUS_TOOL_INCLUDE_DIR;
 #undef GPLUSPLUS_TOOL_INCLUDE_DIR
@@ -128,3 +128,3 @@
 #ifdef GPLUSPLUS_BACKWARD_INCLUDE_DIR
-char cygwin_gplusplus_backward_include_dir[sizeof (GPLUSPLUS_BACKWARD_INCLUDE_DIR) + 1] = GPLUSPLUS_BACKWARD_INCLUDE_DIR;
+char cygwin_gplusplus_backward_include_dir[sizeof (GPLUSPLUS_BACKWARD_INCLUDE_DIR) + 10] = GPLUSPLUS_BACKWARD_INCLUDE_DIR;
 #undef GPLUSPLUS_BACKWARD_INCLUDE_DIR
@@ -137,3 +137,3 @@
 #ifdef LOCAL_INCLUDE_DIR
-char cygwin_local_include_dir[sizeof (LOCAL_INCLUDE_DIR) + 1] = LOCAL_INCLUDE_DIR;
+char cygwin_local_include_dir[sizeof (LOCAL_INCLUDE_DIR) + 10] = LOCAL_INCLUDE_DIR;
 #undef LOCAL_INCLUDE_DIR
@@ -146,3 +146,3 @@
 #ifdef CROSS_INCLUDE_DIR
-char cygwin_cross_include_dir[sizeof (CROSS_INCLUDE_DIR) + 1] = CROSS_INCLUDE_DIR;
+char cygwin_cross_include_dir[sizeof (CROSS_INCLUDE_DIR) + 10] = CROSS_INCLUDE_DIR;
 #undef CROSS_INCLUDE_DIR
@@ -155,3 +155,3 @@
 #ifdef TOOL_INCLUDE_DIR
-char cygwin_tool_include_dir[sizeof (TOOL_INCLUDE_DIR) + 1] = TOOL_INCLUDE_DIR;
+char cygwin_tool_include_dir[sizeof (TOOL_INCLUDE_DIR) + 10] = TOOL_INCLUDE_DIR;
 #undef TOOL_INCLUDE_DIR
@@ -162,3 +162,3 @@
 #define STANDARD_INCLUDE_DIR "/usr/include"
-char cygwin_standard_include_dir[sizeof (STANDARD_INCLUDE_DIR) + 1] = STANDARD_INCLUDE_DIR;
+char cygwin_standard_include_dir[sizeof (STANDARD_INCLUDE_DIR) + 10] = STANDARD_INCLUDE_DIR;
 #undef STANDARD_INCLUDE_DIR
