--- /usr/local/cygwin/lib/gcc-lib/i686-pc-cygwin/2.95.2-6/specs.orig	Mon Jan  1 23:19:00 2001
+++ /usr/local/cygwin/lib/gcc-lib/i686-pc-cygwin/2.95.2-6/specs	Sat Jan  6 21:08:56 2001
@@ -7,3 +7,3 @@
 *cpp:
--remap %(cpp_cpu) %{posix:-D_POSIX_SOURCE}   %{!mno-cygwin:-D__CYGWIN32__ -D__CYGWIN__ -Dunix -D__unix__ -D__unix     -isystem /usr/local/include     -idirafter /usr/include}   %{mno-win32: %{mno-cygwin: %emno-cygwin and mno-win32 are not compatible}}   %{mno-cygwin:-DWIN32 -D__WIN32__ -D__MINGW32__=0.2     %{mthreads:-D_MT}     -isystem /usr/local/include/mingw     -idirafter /usr/include/mingw     -iwithprefixbefore ../../../../mingw/include/g++-3     -iwithprefixbefore ../../../../mingw/include     -iwithprefixbefore ../../../../mingw32/include/g++-3     -iwithprefixbefore ../../../../mingw32/include }   %{!mno-win32:-D_WIN32 -DWINNT -idirafter /usr/include/w32api}
+-remap %(cpp_cpu) %{posix:-D_POSIX_SOURCE}   %{!mno-cygwin:-D__CYGWIN32__ -D__CYGWIN__ -Dunix -D__unix__ -D__unix     -isystem /usr/local/cygwin/include     -idirafter /usr/cygwin/include}   %{mno-win32: %{mno-cygwin: %emno-cygwin and mno-win32 are not compatible}}   %{mno-cygwin:-DWIN32 -D__WIN32__ -D__MINGW32__=0.2     %{mthreads:-D_MT}     -isystem /usr/local/cygwin/i686-pc-cygwin/include/mingw     -idirafter /usr/include/mingw     -iwithprefixbefore ../../../../mingw/include/g++-3     -iwithprefixbefore ../../../../mingw/include     -iwithprefixbefore ../../../../mingw32/include/g++-3     -iwithprefixbefore ../../../../mingw32/include }   %{!mno-win32:-D_WIN32 -DWINNT -idirafter /usr/local/cygwin/i686-pc-cygwin/include/w32api}
 
@@ -46,3 +46,3 @@
 *multilib:
-. ;
+../../../../i686-pc-cygwin/lib/mingw mno-cygwin;
 
@@ -55,3 +55,3 @@
 *multilib_matches:
-
+mno-cygwin mno-cygwin;
 
